tests/unit/data/gomod-mocks/non-vendored/go_mod_download.json

```diff
@@ -0,0 +1,200 @@
+{
+	"Path": "github.com/Microsoft/go-winio",
+	"Version": "v0.6.0",
+	"Info": "/home/acmiel/go/pkg/mod/cache/download/github.com/!microsoft/go-winio/@v/v0.6.0.info",
```

Comment:
I know this is only supposed to be mock data for the tests, but I'm worried that everytime we need to re-generate it, the path will change depending on the person/user who did it.

Reply:
Good point, I didn't notice that. Should be easy enough to fix.

Reply:
There was another problem with `Dir` and `GoMod` in `go list -deps -json` output, especially if you want to use `Dir` later

```
"Dir": "/home/acmiel/RedHat/Cachito/cachi2/tests/unit/data/gomod-mocks/gomod-pandemonium",
```

Fixed by setting both GOMODCACHE and the repo path to a tmpdir and replacing the tmpdir with `{tmpdir}` in case someone needs to replace it later

Reply:
Or maybe better, one `{gomodcache_dir}` placeholder and one `{repo_dir}` placeholder